pageserver/src/branches.rs

```diff
@@ -104,19 +104,28 @@ pub fn create_repo(
     // and we failed to run initdb again in the same directory. This has been solved for the
     // rapid init+start case now, but the general race condition remains if you restart the
     // server quickly.
-    let storage = crate::rocksdb_storage::RocksObjectStore::create(conf, &tenantid)?;
-
-    let repo = crate::object_repository::ObjectRepository::new(
-        conf,
-        std::sync::Arc::new(storage),
-        wal_redo_manager,
-        tenantid,
-    );
+    let repo: Arc<dyn Repository + Sync + Send> = match conf.repository_format {
```

Comment:
There is the same code for creating object repository in repository.rs.
Is there any reason for such code duplication?